services/venice-router/src/main/java/com/linkedin/venice/router/stats/RouterHttpRequestStats.java

```diff
@@ -137,21 +138,34 @@ public class RouterHttpRequestStats extends AbstractVeniceHttpStats {
 
   // QPS metrics
   public RouterHttpRequestStats(
-      VeniceMetricsRepository metricsRepository,
+      MetricsRepository metricsRepository,
       String storeName,
       String clusterName,
       RequestType requestType,
       ScatterGatherStats scatterGatherStats,
       boolean isKeyValueProfilingEnabled) {
     super(metricsRepository, storeName, requestType);
-    emitOpenTelemetryMetrics = metricsRepository.getVeniceMetricsConfig().emitOtelMetrics();
-    openTelemetryMetricFormat = metricsRepository.getVeniceMetricsConfig().getMetricNamingFormat();
-    VeniceOpenTelemetryMetricsRepository otelRepository = metricsRepository.getOpenTelemetryMetricsRepository();
-    commonMetricDimensions = Attributes.builder()
-        .put(getDimensionName(VENICE_STORE_NAME), storeName)
-        .put(getDimensionName(VENICE_REQUEST_METHOD), requestType.name().toLowerCase())
-        .put(getDimensionName(VENICE_CLUSTER_NAME), clusterName)
-        .build();
+    VeniceOpenTelemetryMetricsRepository otelRepository;
+    if (metricsRepository instanceof VeniceMetricsRepository) {
+      VeniceMetricsRepository veniceMetricsRepository = (VeniceMetricsRepository) metricsRepository;
+      emitOpenTelemetryMetrics = (veniceMetricsRepository != null)
```

Comment:
Note that this is impossible...

`null instanceof AnyType` is always false.

An `instanceof` check is also a null check.

Reply:
TIL. Also missed the editor's suggestion. Thanks.